test/widget-sync-integration.test.ts

```diff
@@ -0,0 +1,200 @@
+/**
+ * Widget Synchronization Integration Tests
+ *
+ * Tests for the GitHub #91 comment issue where widgets show different years
+ * after date changes. This tests the integration between time converter
+ * and widgets when world creation timestamps are involved.
+ */
+
+import { describe, it, expect, beforeEach, vi, beforeAll } from 'vitest';
+import { CalendarEngine } from '../src/core/calendar-engine';
+import { TimeConverter } from '../src/core/time-converter';
+import type { SeasonsStarsCalendar } from '../src/types/calendar-types';
+import { setupPF2eEnvironment } from './setup';
+// Import PF2e integration to register the world creation timestamp hook
+import '../src/integrations/pf2e-integration';
+
+describe('Widget Synchronization Integration', () => {
+  let golarionCalendar: SeasonsStarsCalendar;
+  let engine: CalendarEngine;
+  let timeConverter: TimeConverter;
+
+  beforeAll(() => {
+    // Set up PF2e environment using enhanced foundry-test-utils
+    setupPF2eEnvironment({
+      worldCreationTimestamp: new Date('2025-01-01T00:00:00.000Z').getTime() / 1000,
+      currentWorldTime: 0,
+      expectedWorldCreationYear: 2025,
+    });
+
+    // Add other needed globals for the test
+    global.game.user = { isGM: true };
+  });
+
+  beforeEach(() => {
+    // Golarion calendar definition (same as our year calculation tests)
+    golarionCalendar = {
+      id: 'golarion-pf2e',
+      name: 'Golarion Calendar System',
+      months: [
+        { name: 'Abadius', days: 31, description: '' },
+        { name: 'Calistril', days: 28, description: '' },
+        { name: 'Pharast', days: 31, description: '' },
+        { name: 'Gozran', days: 30, description: '' },
+        { name: 'Desnus', days: 31, description: '' },
+        { name: 'Sarenith', days: 30, description: '' },
+        { name: 'Erastus', days: 31, description: '' },
+        { name: 'Arodus', days: 31, description: '' },
+        { name: 'Rova', days: 30, description: '' },
+        { name: 'Lamashan', days: 31, description: '' },
+        { name: 'Neth', days: 30, description: '' },
+        { name: 'Kuthona', days: 31, description: '' },
+      ],
+      weekdays: [
+        { name: 'Moonday', abbreviation: 'Mo' },
+        { name: 'Toilday', abbreviation: 'To' },
+        { name: 'Wealday', abbreviation: 'We' },
+        { name: 'Oathday', abbreviation: 'Oa' },
+        { name: 'Fireday', abbreviation: 'Fi' },
+        { name: 'Starday', abbreviation: 'St' },
+        { name: 'Sunday', abbreviation: 'Su' },
+      ],
+      year: {
+        epoch: 2700,
+        suffix: ' AR',
+      },
+      leapYear: {
+        rule: 'custom',
+        interval: 4,
+        month: 'Calistril',
+        extraDays: 1,
+      },
+      time: {
+        hoursInDay: 24,
+        minutesInHour: 60,
+        secondsInMinute: 60,
+      },
+      intercalary: [],
+    };
```

Comment:
Same here, could just use the JSON file. Why create potential bugs by copying when you don't need to?